spacepy/plot/utils.py

```diff
@@ -294,8 +294,11 @@ def analyze(self):
         if self.interval is None:
             (left, right) = self.ax.get_xaxis().get_view_interval()
             if self._x_is_datetime:
-                right = matplotlib.dates.num2date(right)
-                left = matplotlib.dates.num2date(left)
+                right = matplotlib.dates.num2date(right, tz=self._xdata[0].tzinfo)
+                left = matplotlib.dates.num2date(left, tz=self._xdata[0].tzinfo)
+                if not self._xdata[0].tzinfo:
+                    right = right.replace(tzinfo=None)
+                    left = left.replace(tzinfo=None)
```

Comment:
```suggestion
```

Reply:
This one got marked as a multi-line edit so can't be applied, added as another comment and resolving this.